statistics/handle/handle_hist.go

```diff
@@ -330,6 +346,22 @@ func (h *Handle) writeToChanWithTimeout(taskCh chan *NeededColumnTask, task *Nee
 	return nil
 }
 
+// writeToResultChan safe-writes with panic-recover so one write-fail will not have big impact.
+func (h *Handle) writeToResultChan(resultCh chan model.TableColumnID, rs model.TableColumnID) {
+	defer func() {
+		if r := recover(); r != nil {
+			buf := make([]byte, 4096)
+			stackSize := runtime.Stack(buf, false)
+			buf = buf[:stackSize]
+			logutil.BgLogger().Error("writeToResultChan panicked", zap.Any("error", r), zap.String("stack", string(buf)))
+		}
+	}()
+	select {
+	case resultCh <- rs:
+	default:
```

Comment:
Here we make the sending non-blocking. Do we try to avoid certain blocking case? The result channel seems have enough buffer.

Reply:
The result channel seems have enough buffer.  -- yes, but sending in a safe way so that it won't impact other sqls' tasks if any edge case is missed.